lib/Runtime/Library/TypedArray.cpp

```diff
@@ -143,7 +143,11 @@ namespace Js
             {
                 if (JavascriptOperators::IsObject(firstArgument))
                 {
-                    if (JavascriptOperators::IsIterable(RecyclableObject::FromVar(firstArgument), scriptContext))
+                    if (JavascriptOperators::IsIterable(RecyclableObject::FromVar(firstArgument), scriptContext) &&
+                        (!JavascriptArray::Is(firstArgument) ||
+                        JavascriptOperators::HasOwnProperty(firstArgument, PropertyIds::_symbolIterator, scriptContext) ||
+                        DynamicObject::FromVar(firstArgument)->GetPrototype() != scriptContext->GetLibrary()->GetArrayPrototype() ||
+                        scriptContext->GetLibrary()->GetArrayObjectHasUserDefinedIterator()))
```

Comment:
I looked through `IsIterable()` and it only calls `JavascriptOperators::GetProperty()` or `::HasProperty()`, not doing any calling. Can't see any side effects.

Nevertheless, simplifying makes sense esp. since we don't mind property look-up here. Updated in the latest rev. Thanks!